pmda/simple.py

```diff
@@ -0,0 +1,126 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# PMDA
+# Copyright (c) 2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+"""
+Parallel Analysis helper --- :mod:`pmda.simple`
+===============================================
+
+Collection of decorators to analyze
+
+"""
+from __future__ import absolute_import
+
+from MDAnalysis.core.groups import AtomGroup
+from MDAnalysis.core.universe import Universe
+from MDAnalysis.coordinates.base import ProtoReader
+import numpy as np
+
+from .parallel import ParallelAnalysisBase
+
+
+class AnalysisFromFunction(ParallelAnalysisBase):
+    """
+    Create an analysis from a function working on AtomGroups
+
+    Attributes
+    ----------
+    results : ndarray
+        results of calculation are stored after call to ``run``
+
+    Example
+    -------
+    >>> def rotation_matrix(mobile, ref):
+    >>>     return mda.analysis.align.rotation_matrix(mobile, ref)[0]
+
+    >>> rot = AnalysisFromFunction(rotation_matrix,
+                                   trajectory, mobile, ref).run()
+    >>> print(rot.results)
+
+    Raises
+    ------
+    ValueError : if ``function`` has the same kwargs as ``BaseAnalysis``
+    """
+
+    def __init__(self, function, universe, *args, **kwargs):
+        """
+        Parameters
+        ----------
+        function : callable
+            function to evaluate at each frame
+        Universe : :class:`~MDAnalysis.core.groups.Universe`
+            a :class:`MDAnalysis.core.groups.Universe` (the
+            `atomgroups` must belong to this Universe)
+        *args : list
+           arguments for ``function``
+        **kwargs : dict
+           keyword arguments for ``function``. keyword arguments with name
+           'universe' or 'atomgroups' will be ignored! Atomgroups to analyze
+           can not be passed as keyword arguments currently.
+        """
+
+        self.function = function
+
+        # collect all atomgroups with the same trajectory object as universe
+        trajectory = universe.trajectory
+        arg_ags = []
+        self.other_args = []
+        for arg in args:
+            if isinstance(arg,
+                          AtomGroup) and arg.universe.trajectory == trajectory:
+                arg_ags.append(arg)
+            else:
+                self.other_args.append(arg)
+
+        super(AnalysisFromFunction, self).__init__(universe, arg_ags)
+        self.kwargs = kwargs
+
+    def _prepare(self):
+        self.results = []
+
+    def _single_frame(self, ts, atomgroups):
+        if len(self.other_args) != 0:
+            atomgroups.extend(*self.other_args)
+        return self.function(*atomgroups, **self.kwargs)
+
+    def _conclude(self):
+        self.results = np.hstack(self._results)
```

Comment:
Are you sure that `hstack` is correct? For the RMSD class I had to change it to `vstack`? https://github.com/MDAnalysis/pmda/blob/006a4f3779ca89c728e98d69b082a6061491932d/pmda/rms.py#L76 to get a continuous time series.

Reply:
It's weird that you needed the change to be honest. The `hstack` does the right thing. I'm also testing this. For a trajectory of 49 frames the results array has shape (49, N) with N being the number of return values from the function used.